File: src/main/java/com/neo/caption/ocr/service/TesseractService.java
package com.neo.caption.ocr.service;

import com.neo.caption.ocr.domain.dto.TessLanguageDto;
import org.bytedeco.tesseract.TessBaseAPI;

import java.util.List;

public interface TesseractService {

    TessBaseAPI initTessBaseApi(String language);

    List<TessLanguageDto> getTessLanguageDtoList();

}

File: src/main/kotlin/com/neo/caption/ocr/service/TesseractService.kt
package com.neo.caption.ocr.service

import com.neo.caption.ocr.domain.entity.TesseractConfig
import com.neo.caption.ocr.domain.vo.TesseractConfigVo
import com.neo.caption.ocr.domain.vo.TesseractOptionVo
import org.bytedeco.tesseract.TessBaseAPI

interface TesseractService {

    val supportedLanguage: List<String>

    fun getDefaultConfig(): TesseractConfigVo

    fun getTesseractOption(): TesseractOptionVo

    fun initTessBaseApi(tesseractConfig: TesseractConfig): TessBaseAPI

}